Sources/Features/Client/Model/AttachmentsStorage.swift

```diff
@@ -0,0 +1,129 @@
+        let attachmentBookmarks = try convertAttachmentUrlsToBookmarks(attachments)
+        
+        if #available(iOS 11.0, tvOS 11.0, macOS 10.13, *) {
+            try (attachmentBookmarks as NSDictionary).write(to: config.fileUrl)
+        } else {
+            guard (attachmentBookmarks as NSDictionary).write(to: config.fileUrl, atomically: true) else {
+                throw FileError.fileNotWritten
+            }
+        }
+        BacktraceLogger.debug("Stored attachments at path: \(config.fileUrl)")
+    }
+    
+    static func retrieve(fileName: String) throws -> Attachments {
+        let config = try Config(fileName: fileName)
+        guard FileManager.default.fileExists(atPath: config.fileUrl.path) else {
+            throw FileError.fileNotExists
+        }
+        // load file to NSDictionary
+        let dictionary: NSDictionary
+        if #available(iOS 11.0, tvOS 11.0, macOS 10.13, *) {
+            dictionary = try NSDictionary(contentsOf: config.fileUrl, error: ())
+        } else {
+            guard let dictionaryFromFile = NSDictionary(contentsOf: config.fileUrl) else {
+                throw FileError.invalidPropertyList
+            }
+            dictionary = dictionaryFromFile
+        }
+        
+        guard let bookmarks = dictionary as? Bookmarks else {
+            BacktraceLogger.debug("Could not convert stored dictionary to Bookmarks type")
+            throw FileError.unsupportedScheme
+        }
+        guard let attachments = try? extractAttachmentUrls(bookmarks) else {
+            BacktraceLogger.debug("Could not extract attachment URLs from stored attachments Bookmarks")
+            throw FileError.invalidPropertyList
+        }
+
+        BacktraceLogger.debug("Retrieved attachments from path: \(config.fileUrl)")
+        return attachments
+    }
+    
+    static func retrieveAsArray(fileName: String) throws -> [String] {
+        guard let attachments = try? retrieve(fileName: fileName) else {
+            return []
+        }
+        var attachmentsArray = [ String ]()
+        for attachment in attachments {
+            attachmentsArray.append(attachment.value.path)
+        }
+        return attachmentsArray
+    }
+    
+    static func remove(fileName: String) throws {
+        let config = try Config(fileName: fileName)
+        // check file exists
+        guard FileManager.default.fileExists(atPath: config.fileUrl.path) else {
+            throw FileError.fileNotExists
+        }
+        // remove file
+        try FileManager.default.removeItem(at: config.fileUrl)
+        BacktraceLogger.debug("Removed attachments plist at path: \(config.fileUrl)")
+    }
+    
+    private static func convertAttachmentUrlsToBookmarks(_ attachments: Attachments) throws -> Bookmarks {
+        var attachmentsBookmarksDict = Bookmarks()
+        for attachment in attachments {
+            do { let bookmark = try attachment.value.bookmarkData(options: URL.BookmarkCreationOptions.minimalBookmark)
```

Comment:
Expression should start from the new line

Reply:
Will fix, thank you!